<launch>
  <!-- setting -->
  <rosparam param="wheelbase"> 2.05 </rosparam>
  <rosparam param="stop_threshold"> 0.1 </rosparam>
  <rosparam param="wheel_steering_weight"> 4.0 </rosparam>

  <!-- Real Multisense visual odometry -->
  <include file="$(find drc_task_common)/launch/real_multisense_viso2.launch" />
  
  <!-- verification using visual odometry -->
  <node name="visual_odometry_verification" pkg="drc_task_common" type="visual_odometry_verification" output="screen" >
    <remap from="odometry" to="/viso_node/odometry" />
    <remap from="steering" to="/vo_verification/steering_angle" />
    <remap from="velocity" to="/vo_verification/linear_velocity" />
    <remap from="angular_velocity" to="/vo_verification/angular_velocity" />
  </node>

  <!-- visualization of steering angle -->
  <node name="steering_angle_throttle" pkg="topic_tools" type="throttle" output="screen" 
        args="messages /vo_verification/steering_angle 2.0 /steering_throttle"
        />
  <node name="handle_viewer" pkg="gazebo_drive_simulator" type="SteeringWheelViewer.py" output="screen" >
    <remap from="steering" to="/steering_throttle" />
  </node>
  
</launch>